qpsolvers_benchmark/test_sets/maros_meszaros_dense_posdef.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+#
+# Copyright 2022 Stéphane Caron
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Dense subset with positive definite P of the Maros-Meszaros test set."""
+
+from typing import Iterator
+
+from ..problem import Problem
+from .maros_meszaros import MarosMeszaros
+import numpy as np
+
+
+
+class MarosMeszarosDensePosdef(MarosMeszaros):
+    """Subset of Maros-Meszaros restricted to smaller dense problems."""
+
+    @property
+    def description(self) -> str:
+        """Description of the test set."""
+        return (
+            "Subset of the Maros-Meszaros test set "
+            "restricted to smaller dense problems "
+            "with positive definite Hessian matrix."
+        )
+
+    @property
+    def title(self) -> str:
+        """Test set title."""
+        return "Maros-Meszaros dense positive definite subset"
+
+    @property
+    def sparse_only(self) -> bool:
+        """Test set is dense."""
+        return False
+
+    @staticmethod
+    def count_constraints(problem: Problem):
+        """Count inequality and equality constraints.
+
+        Notes:
+            We only count box inequality constraints once, and only from lower
+            bounds. That latter part is specific to this test set.
+        """
+        m = 0
+        if problem.G is not None:
+            m += problem.G.shape[0]
+        if problem.A is not None:
+            m += problem.A.shape[0]
+        if problem.lb is not None:
+            m += problem.lb.shape[0]
+        return m
+
+
+    @staticmethod
+    def is_pos_def(x):
+        return np.all(np.linalg.eigvals(x) > 0)
```

Comment:
Just did a quick comparison with:

```python
def is_posdef_chol(M):
    try:
        np.linalg.cholesky(M)
    except np.linalg.LinAlgError:
        return False
    return True
```

The results on my machine are:

```python
In [16]: M = np.random.random((100, 100))

In [17]: %timeit is_pos_def(M)
7.65 ms ± 387 µs per loop (mean ± std. dev. of 7 runs, 100 loops each)

In [18]: %timeit is_posdef_chol(M)
22.9 µs ± 695 ns per loop (mean ± std. dev. of 7 runs, 10000 loops each)
```

Reply:
Noted. Thanks.